### PR TITLE
Initial WebRTC support

### PR DESCRIPTION
**Pull request by Faless**
This PR is the result of the joint effort of me and @brandonmakin during last GSoC.
Sadly, due to build issues and size, non-HTML5 support (on which @brandonmakin worked hard last summer) will be delegated to a GDNative plugin (and which will soon be available here https://github.com/godotengine/webrtc-native/ ).
This PR do not yet include the high level multiplayer peer (which might be added to this PR in a future commit as soon as we stabilize it a bit).

This will, of course, all come in `3.2`.

### CODE REVIEW

**Comment by reduz** [comment]
up to you, @akien-mga

**Comment by akien-mga** [comment]
Looks good to me, but @karroffel might want to check the GDNative changes and @eska014 have a look at the JS integration.

**Comment by Faless** [comment]
There's one little change I need to push (will do later today), to allow compiling with `module_gdnative_enabled=no`. Plus I have to remove the google demo stun server leftover from JS :-/

**Comment by Faless** [comment]
@eska014 have a look at: https://github.com/godotengine/godot/pull/26792/files#diff-8cd4b6695db965b42410d87a3c6069d9 , maybe we can reuse it for XMLHTTPRequests too? I saw you had something similar using an array.

File: modules/gdnative/gdnative_api.json [hunk 1]
```diff
@@ -6452,6 +6452,14 @@
 		  ["godot_object *", "p_obj"],
 		  ["const godot_net_multiplayer_peer *", "p_interface"]
           ]
+        },
+        {
+          "name": "godot_net_bind_webrtc_peer",
```
Comment (karroffel) [inline modules/gdnative/gdnative_api.json]: This technically does not break binary compatibility, but the API struct for this extension will be "valid" even with Godot versions where this function does not exist.

So this should be in the "next" extension of the "NET" extension.

**Review by karroffel — CHANGES_REQUESTED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**